runtime/src/mailbox_api.rs

```diff
@@ -0,0 +1,237 @@
+        match self {
+            MailboxResp::Header(resp) => resp.as_bytes_mut(),
+            MailboxResp::GetIdevCsr(resp) => resp.as_bytes_mut(),
+            MailboxResp::GetLdevCert(resp) => resp.as_bytes_mut(),
+            MailboxResp::StashMeasurement(resp) => resp.as_bytes_mut(),
+            MailboxResp::InvokeDpeCommand(resp) => resp.as_bytes_mut(),
+            MailboxResp::TestGetFmcAliasCert(resp) => resp.as_bytes_mut(),
+            MailboxResp::FipsVersion(resp) => resp.as_bytes_mut(),
+        }
+    }
+
+    /// Calculate and set the checksum for a response payload
+    /// Takes into account the size override for variable-lenth payloads
+    pub fn populate_chksum(&mut self) -> CaliptraResult<()> {
+        // Calc checksum, use the size override if provided
+        let checksum = caliptra_common::checksum::calc_checksum(0, &self.as_bytes()[4..]);
+
+        // cast as header struct
+        let hdr: &mut MailboxRespHeader =
+            LayoutVerified::<&mut [u8], MailboxRespHeader>::new(&mut self.as_bytes_mut()[..core::mem::size_of::<MailboxRespHeader>()])
+            .ok_or(CaliptraError::RUNTIME_INSUFFICIENT_MEMORY)?
+            .into_mut();
+
+        // Set the chksum field
+        hdr.chksum = checksum;
+
+        Ok(())
+    }
+}
+
+impl Default for MailboxResp {
+    fn default() -> Self {
+        MailboxResp::Header(MailboxRespHeader::default())
+    }
+}
+
+// HEADER
+#[repr(C)]
+#[derive(Debug, AsBytes, FromBytes, PartialEq, Eq)]
+pub struct MailboxReqHeader {
+    pub chksum: i32,
+}
+
+#[repr(C)]
+#[derive(Debug, AsBytes, FromBytes, PartialEq, Eq)]
+pub struct MailboxRespHeader {
+    pub chksum: i32,
+    pub fips_status: u32,
+}
+
+impl MailboxRespHeader {
+    pub const FIPS_STATUS_APPROVED: u32 = 0;
+}
+
+impl Default for MailboxRespHeader {
+    fn default() -> Self {
+        Self {
+            chksum: 0,
+            fips_status: MailboxRespHeader::FIPS_STATUS_APPROVED,
+        }
+    }
+}
+
+// GET_IDEV_CSR
+// No command-specific input args
+#[repr(C)]
+#[derive(Debug, AsBytes, FromBytes, PartialEq, Eq)]
+pub struct GetIdevCsrResp {
+    pub hdr: MailboxRespHeader,
+    pub data_size: u32,
+    pub data: [u8; GetIdevCsrResp::DATA_MAX_SIZE], // variable length
+}
+impl GetIdevCsrResp {
+    pub const DATA_MAX_SIZE: usize = 1024;
+}
+
+// GET_LDEV_CERT
+// No command-specific input args
+#[repr(C)]
+#[derive(Debug, AsBytes, FromBytes, PartialEq, Eq)]
+pub struct GetLdevCertResp {
+    pub hdr: MailboxRespHeader,
+    pub data_size: u32,
+    pub data: [u8; GetLdevCertResp::DATA_MAX_SIZE], // variable length
+}
+impl GetLdevCertResp {
+    pub const DATA_MAX_SIZE: usize = 1024;
+}
+
+// ECDSA384_SIGNATURE_VERIFY
+#[repr(C)]
+#[derive(Debug, AsBytes, FromBytes, PartialEq, Eq)]
+pub struct EcdsaVerifyCmdReq {
+    pub hdr: MailboxReqHeader,
+    pub pub_key_x: [u8; 48],
+    pub pub_key_y: [u8; 48],
+    pub signature_r: [u8; 48],
+    pub signature_s: [u8; 48],
+}
+// No command-specific output args
```

Comment:
Shouldn't this return a FIPS status?

Reply:
This comment may need to change. What I was trying to convey is that it has not additional fields specific to the command. I could either
1. Make the comment clearer, like "Resp header is the only output" or any other suggestion
2. (I don't think we want to do this) create req and resp structs for these cases that only contain the header

Reply:
Oh right, I forgot we just return the header. For consistency, maybe we should have request/response structures for every command, even if they only field is the header. wdyt?

Reply:
Yeah that would be fine.  The minor downside to that is then we do need to actually put an entry in the resp enum (and in as_bytes) for each type instead of just using generic "header" in there if we want to actually use them in the runtime code.

Or we can add them for clarity but continue to use the generic "header" in the enum, but that may be more confusing.

Just a tradeoff of consistency for a little redundancy. Whichever you prefer is good with me.

Reply:
Hmm. Ok, go ahead and leave it as you have it for now.